src/Tasks/DownloadFile.cs

```diff
@@ -150,7 +150,11 @@ private async Task DownloadAsync(Uri uri, CancellationToken cancellationToken)
                     {
                         // HttpRequestException does not have the status code so its wrapped and thrown here so that later on we can determine
                         // if a retry is possible based on the status code
+#if RUNTIME_TYPE_NETCORE
+                        throw new HttpRequestException(e.Message, e.InnerException, response.StatusCode);
```

Comment:
This constructor doesn't exist in the framework flavor

Reply:
Make this comment permanent IMO